components/src/components/molecules/Dropdown/Dropdown.tsx

```diff
@@ -440,58 +501,108 @@ export const Dropdown = ({
     return () => {
       document.removeEventListener('mousedown', handleClickOutside)
     }
-  }, [dropdownRef, isOpen, setIsOpen])
-
-  const button: React.ReactNode = children ? (
-    React.Children.map(children, (child) => {
-      if (!React.isValidElement(child)) return null
-      return React.cloneElement(child as any, {
-        ...buttonPropsWithIndicator,
-        zindex: '10',
-        pressed: isOpen ? 'true' : undefined,
-        onClick: () => setIsOpen((prev) => !prev),
-        ref: buttonRef,
-      })
-    })
-  ) : (
-    <Button
-      data-testid="dropdown-btn"
-      pressed={isOpen}
-      ref={buttonRef}
-      suffix={chevron && <Chevron $direction={direction} $open={isOpen} />}
-      width="fit"
-      onClick={() => setIsOpen((prev) => !prev)}
-      {...buttonPropsWithIndicator}
-    >
-      {label}
-    </Button>
-  )
+  }, [dropdownRef, isOpen, setIsOpen, buttonRef, actionSheetRef])
+}
+
+export const Dropdown = ({
+  children,
+  buttonProps,
+  items = [],
+  chevron = true,
+  align = 'left',
+  menuLabelAlign,
+  width = 150,
+  mobileWidth = width,
+  shortThrow = false,
+  keepMenuOnTop = false,
+  label,
+  direction = 'down',
+  isOpen: _isOpen,
+  setIsOpen: _setIsOpen,
+  indicatorColor,
+  responsive = true,
+  cancelLabel = 'Cancel',
+  ...props
+}: Props & (PropsWithIsOpen | PropsWithoutIsOpen)) => {
+  const dropdownRef = React.useRef<any>()
+  const buttonRef = React.useRef<HTMLButtonElement>(null)
+  const actionSheetRef = React.useRef<HTMLDivElement>(null)
+  const internalOpen = React.useState(false)
+  const [isOpen, setIsOpen] = _setIsOpen ? [_isOpen, _setIsOpen] : internalOpen
+
+  useClickOutside(dropdownRef, buttonRef, actionSheetRef, setIsOpen, isOpen)
+  const screenSize = useScreenSize()
 
   return (
     <>
-      {button}
-      <DynamicPopover
-        additionalGap={-10}
-        align={align === 'left' ? 'start' : 'end'}
-        anchorRef={buttonRef}
-        hideOverflow={!keepMenuOnTop}
-        isOpen={isOpen}
-        mobilePlacement={direction === 'down' ? 'bottom' : 'top'}
-        mobileWidth={mobileWidth}
-        placement={direction === 'down' ? 'bottom' : 'top'}
-        popover={
-          <DropdownMenu
-            direction={direction}
-            items={items}
-            labelAlign={menuLabelAlign}
-            setIsOpen={setIsOpen}
-            shortThrow={shortThrow}
-            {...props}
-            ref={dropdownRef}
-          />
-        }
-        width={width}
+      <DropdownButton
+        {...{
+          children,
+          buttonRef,
+          chevron,
+          direction,
+          isOpen,
+          setIsOpen,
+          label,
+          items,
+          buttonProps,
+          indicatorColor,
+        }}
       />
+      {match({ responsive, screenSize })
```

Comment:
i get why pattern matching is being used here, but not sure that we should use a lib for basic functionality in a single component

Reply:
> i get why pattern matching is being used here, but not sure that we should use a lib for basic functionality in a single component

Will be used more widely going forward

Reply:
It's about 1kb gzipped